### PR TITLE
Add CI

### PR DESCRIPTION
**Pull request by QuentinTorg**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+name: Build MultiSense ROS
+
+on:
+  push:
+    branches:
+      - master
+  release:
+  pull_request:
+    branches:
+      - master
```
Comment (grzleadams) [inline .github/workflows/build.yml:10]: Do you want to add a workflow_dispatch trigger too?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      cancel-in-progress: ${{ github.ref_name != 'main' }}
+    strategy:
+      fail-fast: false
+      matrix:
+        ros_distro:
+          - noetic
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
```
Comment (grzleadams) [inline .github/workflows/build.yml]: Can all of the submodules be checked out without a token?

Comment (QuentinTorg) [reply .github/workflows/build.yml]: yep, this is a public repo. the only submodule is libmultisense, so I can remove the recursive too

File: .github/workflows/build.yml (hunk 1 above)
Comment (QuentinTorg) [reply .github/workflows/build.yml:10]: does that allow manual running of the workflow? Seems like it would be nice to have as an option

Comment (grzleadams) [reply .github/workflows/build.yml:10]: Yeah, I just add it to all workflows unless there's a good reason not to.

**Review by grzleadams — APPROVED** [review]

**Pull request merged**